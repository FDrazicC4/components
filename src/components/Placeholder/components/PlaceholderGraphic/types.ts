

export interface Props {

    /**
     * Classes to be passed through to the component.
     */
    className?: string,

    /**
     * The icon or graphic to be displayed on top.
     */
    graphic?: string,

}
