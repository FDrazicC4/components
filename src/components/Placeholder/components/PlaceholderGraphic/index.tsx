import React, { ReactElement } from 'react';
import Icon from '@components/Icon';
import Typography from '@components/Typography';
import { Props } from './types';
import clsx from 'clsx';


function PlaceholderGraphic({
    className,
    graphic,
}: Props): ReactElement {


    const placeholderClasses = clsx(
        'brew-Placeholder__graphic',
        className,
    );


    return (
        <Typography as="p" className={placeholderClasses} variant="body1">
            <Icon>{graphic}</Icon>
        </Typography>
    );


}


export default PlaceholderGraphic;
