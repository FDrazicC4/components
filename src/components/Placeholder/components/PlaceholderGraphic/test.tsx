import React from 'react';
import renderer from 'react-test-renderer';
import PlaceholderGraphic from './index';
import { shallow } from 'enzyme';


describe('PlaceholderGraphic', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<PlaceholderGraphic graphic="foo" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const placeholderGraphic = shallow(<PlaceholderGraphic graphic="foo" />);
        expect(placeholderGraphic.hasClass('brew-Placeholder__graphic')).toBe(true);
    });

    it('correctly adds a className', () => {
        const className = `class-${Math.floor(Math.random() * 1000)}`;
        const placeholderGraphic = shallow(<PlaceholderGraphic className={className} graphic="foo" />);
        expect(placeholderGraphic.hasClass(className)).toBe(true);
    });

});
