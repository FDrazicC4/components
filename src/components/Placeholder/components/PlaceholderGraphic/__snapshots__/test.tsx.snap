// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderGraphic renders correctly 1`] = `
<p
  className="wrapper variant--body1 color--initial align--inherit display--initial brew-Placeholder__graphic"
>
  <span
    className="wrapper color--inherit size--inherit"
  >
    <svg
      height="100%"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      ‚òê
    </svg>
  </span>
</p>
`;
